docs/fault-proof-alpha/immunifi.md

```diff
@@ -0,0 +1,82 @@
+
+## Bounty Scope
+The scope of the bounty is limited to the fault proof alpha system. This includes the following components, in order of security review priority:
+1. **Cannon**: The [Cannon][cannon] binary and its dependencies.
+1. **op-program**: The [`op-program`][op-program] binary and its dependencies.
+1. **Smart Contracts**
+    1. The [Cannon][cannon-contracts] contracts and their dependencies.
+    1. The [Dispute Game][dispute-game] and their dependencies.
+1. **op-challenger**: The [`op-challenger`][op-challenger] binary and its dependencies.
+
+As mentioned above in the "[Plans for the next iteration](#plans-for-the-next-iteration)" section, there will soon be a number of large architectural changes to the [dispute smart contracts][dispute-game]
+as well as the [`op-challenger`][op-challenger] in order to support the features that will bring the system to a production ready state. During this time, it is unlikely that [Cannon][cannon], the [Cannon contracts][cannon-contracts],
+or the [`op-program`][op-program] will change significantly, and as such, we recommend focusing efforts primarily on these components.
+
+There are several key invariants that must be maintained in order for the system to be considered secure:
+1. **Cannon**
+    1. [Cannon][cannon]'s `mipsevm` must be functionally equivalent to the [MIPS thread context][cannon-contracts] implemented in Solidity. Any disparities in behavior is considered a bug.
+        1. Both [Cannon][cannon] and the on-chain [MIPS thread context][cannon-contracts] must produce the same output given an identical setup state and input data.
+    1. Both [Cannon][cannon] and the on-chain [MIPS thread context][cannon-contracts] must produce a deterministic output given an identical setup state and input data.
+    1. Both [Cannon][cannon] and the on-chain [MIPS thread context][cannon-contracts] must never panic on a valid state transition.
+    1. The `PreimageOracle` contract's local data storage must not be able to be corrupted by an external party.
+1. **op-program**
+    1. The [`op-program`][op-program] must produce a deterministic output given an identical setup state and input data.
+1. **Dispute Game Contracts**
+    1. Assuming the presence of an `honest challenger` (defined by the behavior of the [`op-challenger`][op-challenger]) participating within the game, the `FaultDisputeGame` utilizing the `MIPS` VM **must always** resolve favorably towards the honest L2 state.
+        1. *Note*: The only exception to this invariant in the alpha dispute game (with a `MIPS` VM) is the aforementioned DoS vector. This is not considered a valid bug report, however any other violation of this invariant is.
+1. **op-challenger**
+    1. The honest `op-challenger` must never make a claim that does not support the honest outcome of the dispute game (i.e., the outcome which favors the honest L2 state being considered canonical).
```

Comment:
Do we need to explicitly exclude cases where the challenger posts a claim that is later invalidated if it doesn't affect the game resolution? ie we don't claim the challenger would never lose its bond (if there were any) or that it doesn't post pointless claims.